### ISSUE TITLE
The input of LDM

### ISSUE THREAD

**Issue by QijingGJ** [LemuelPuglisi/BrLP#12]
Dear Lemuel,

I hope everything is going well with you. I truly appreciate your assistance.

I've reviewed the paper on Latent Diffusion Models (LDM) as well as the BrLP code, and I have a few questions I'd like to ask you for help.

（1）In the LDM paper, z is sampled from the mean and variance, as highlighted in yellow in the following figure. Does this mean that after the image passes through AutoencoderKL to obtain z_mu and z_sigma, is it necessary to calculate z_vae = z_mu + eps * z_sigma to obtain the input for Unet？

 
![Image](https://github.com/user-attachments/assets/685dca49-6590-4fb9-b75c-8e1dd4501b04)

However, in extract_latents.py, mri_latent is the return value of autoencoder.encode. It is z_mu. Why is that？

（2）During the decoding process of latent, the scale_factor in z = z/scale_factor is the same as the scale_factor during training. Then, can we directly specify scale_factor = 0.5？

（3）I want to know how many epochs are generally trained for diffusion unet？    

![Image](https://github.com/user-attachments/assets/b55a1087-e5a4-4248-97b4-9f77cd0cbefd)

The above is the loss curve of my training for 100 epochs. Is this overfitting? However, even after sampling the training set, the results are not exactly the same. What should I do to enhance the generation ability of diffusion unet？

-----------------------------------------------------------------------------------------------------------------
Train set example: The left side is the generated one, and the right side is the original one

 
![Image](https://github.com/user-attachments/assets/8369576e-d129-418e-bf9c-95250b1afb2c)

Valid set example: The left side is the generated one, and the right side is the original one

 
![Image](https://github.com/user-attachments/assets/bda7357c-1c18-4336-bfa4-5bb9b82786b1)

-----------------------------------------------------------------------------------------------------------------

Thank you for taking the time to help me in your busy schedule.
Looking forward to hearing from you, thank you.

Best regards,
JingQi

**Comment by LemuelPuglisi** [comment]
Dear JingQi,

I'll try to address all your questions:

1. In practice, many implementations (see `LatentDiffusionInferer` in [MONAI](https://github.com/Project-MONAI/MONAI/blob/dev/monai/inferers/inferer.py)) skip the step highlighted in yellow and directly use the latent predicted by the VAE.

2. I'm afraid I didn't quite understand your question. Could you please clarify?

3. It does seem like your U-Net is overfitting—could your dataset be too small? How many images are you using?

Also, I see you're trying to compare the generated samples to the real ones. Have you trained a ControlNet to condition the generation on a specific input? Without conditioning, the LDM outputs aren't structurally aligned with any reference.

Best regards,
Lemuel

**Comment by QijingGJ** [comment]
Dear Lemuel,

Regarding the second question, I seem to have a new understanding. scale_factor doesn't need to be very precise, so it only needs to adopt the variance of the first batch. Is my understanding correct？

My dataset has over 1,000 pairs of matching data. I'm not quite sure if it is a small dataset for this model. 

I haven't trained a ControlNet. I have a conditionally guided image C. C is mapped to the latent space zc through a trained AutoencoderKL, and the size of zc is (3,16,16). zc is concatenated together with the denoised zt and input into Diffusion Unet. Is this okay? I want to know if training a ControlNet would be a better choice for image conditions?

I'm extremely grateful for your help and look forward to receiving your reply.

Best regards,
JingQi

**Comment by LemuelPuglisi** [comment]
Hi JingQi,

> Regarding the second question, I seem to have a new understanding. scale_factor doesn't need to be very precise, so it only needs to adopt the variance of the first batch. Is my understanding correct？

Yes, I've seen many implementations that do it this way.

> My dataset has over 1,000 pairs of matching data. I'm not quite sure if it is a small dataset for this model.

Your model would definitely benefit from a larger dataset. However, if 1,000 pairs are all you have, I recommend selecting the model based on validation loss for further analysis.

> I haven't trained a ControlNet. I have a conditionally guided image C. C is mapped to the latent space zc through a trained AutoencoderKL, and the size of zc is (3,16,16). zc is concatenated together with the denoised zt and input into Diffusion Unet. Is this okay? I want to know if training a ControlNet would be a better choice for image conditions?

It really depends on what you're trying to achieve. Is your task disease progression? 

Best regards, 
Lemuel

**Comment by QijingGJ** [comment]
Hello Lemuel,

> Your model would definitely benefit from a larger dataset. However, if 1,000 pairs are all you have, I recommend selecting the model based on validation loss for further analysis

Thank you. I will try your suggestion.


> It really depends on what you're trying to achieve. Is your task disease progression?

My task is not disease progression but modal transformation, such as T1 to T2 in MRI.  I'm not quite sure whether this direct concat can be an effective conditional guidance. I'm very much looking forward to your advice.

Best regards,
JingQi

**Comment by LemuelPuglisi** [comment]
Hi JingQi,

Concatenation should work fine—there are many ways to approach this, including using ControlNet. I’d recommend experimenting a bit to see what works best. Good luck!

Best regards,
Lemuel

**Comment by QijingGJ** [comment]
Hello Lemuel,

I will experiment with more settings. Thank you again for your assistance, and I wish you all the best in your research endeavors.

Best regards,
JingQi
